cmd/geth/simulatedbeacon/simulated_beacon.go

```diff
@@ -0,0 +1,240 @@
+			log.Error("failed to initiate PoS transition for genesis via Forkchoiceupdated", "err", err)
+			return
+		}
+	}
+
+	beginSealing := func() (engine.ForkChoiceResponse, error) {
+		tstamp := uint64(time.Now().Unix())
+		if tstamp <= lastBlockTime {
+			tstamp = lastBlockTime + 1
+		}
+		if c.pendingWithdrawals == nil {
+			c.pendingWithdrawals = c.withdrawals.pop()
+		}
+		return engineAPI.ForkchoiceUpdatedV2(curForkchoiceState, &engine.PayloadAttributes{
+			Timestamp:             tstamp,
+			SuggestedFeeRecipient: c.getFeeRecipient(),
+			Withdrawals:           c.pendingWithdrawals,
+		})
+	}
+	finalizeSealing := func(id *engine.PayloadID, onDemand bool) error {
+		payload, err := engineAPI.GetPayloadV1(*id)
+		if err != nil {
+			return fmt.Errorf("error retrieving payload: %v", err)
+		}
+
+		if onDemand && len(payload.Transactions) == 0 && len(payload.Withdrawals) == 0 {
+			// If the payload is empty, despite there being pending transactions,
+			// that indicates that we need to give it more time to build the block.
+			if buildWaitTime < 10*time.Second {
+				buildWaitTime += buildWaitTime
+			}
+			// TODO: don't lose withdrawals
+			return nil
+		}
+		buildWaitTime = 100 * time.Millisecond // Reset it
+		// mark the payload as canon
+		if _, err = engineAPI.NewPayloadV2(*payload); err != nil {
+			return fmt.Errorf("failed to mark payload as canonical: %v", err)
+		}
+		curForkchoiceState = engine.ForkchoiceStateV1{
+			HeadBlockHash:      payload.BlockHash,
+			SafeBlockHash:      payload.BlockHash,
+			FinalizedBlockHash: payload.BlockHash,
+		}
+		// mark the block containing the payload as canonical
+		if _, err = engineAPI.ForkchoiceUpdatedV2(curForkchoiceState, nil); err != nil {
+			return fmt.Errorf("failed to mark block as canonical: %v", err)
+		}
+		c.pendingWithdrawals = nil
+		lastBlockTime = payload.Timestamp
+		return nil
+	}
+	var fcId *engine.PayloadID
+	if fc, err := beginSealing(); err != nil {
+		log.Error("Error starting sealing-work", "err", err)
+		return
+	} else {
+		fcId = fc.PayloadID
+	}
+	onDemand := (c.period == 0)
+	for {
+		select {
+		case <-c.shutdownCh:
+			return
+		case <-ticker.C:
+			if onDemand {
+				// Do nothing as long as blocks are empty
+				if pendingTxs, _ := c.eth.APIBackend.TxPool().Stats(); pendingTxs == 0 && len(c.pendingWithdrawals) == 0 {
+					ticker.Reset(buildWaitTime)
+					continue
+				}
+			}
+			if err := finalizeSealing(fcId, onDemand); err != nil {
+				log.Error("Error collecting sealing-work", "err", err)
+				return
+			}
+			if fc, err := beginSealing(); err != nil {
+				log.Error("Error starting sealing-work", "err", err)
+				return
+			} else {
+				fcId = fc.PayloadID
+			}
+			if !onDemand {
+				ticker.Reset(time.Second * time.Duration(c.period))
+			} else {
+				ticker.Reset(buildWaitTime)
+			}
+		}
+	}
+}
+
+func RegisterAPIs(stack *node.Node, c *SimulatedBeacon) {
+	stack.RegisterAPIs([]rpc.API{
+		{
+			Namespace: "dev",
```

Comment:
No there was no discussion.  I'm fine to move it under `debug` and not add a new namespace is this is what we would like.

Reply:
I'd vote for adding the `dev` namespace or something like it. Because the plan was at some point to standardize the `debug` namespace. IMO we should at least avoid adding new methods there.